### DISCUSSION TITLE
Error: unsupported MediaType: "application/vnd.docker.distribution.manifest.v1+prettyjws"

### DISCUSSION THREAD

**Discussion by sanketg86** [aquasecurity/starboard#609] [Questions & Help] [answered]
am getting below error in startboard-operator.

GKE cluster version: v1.19.9-gke.1400
Starboard-Operator version: v0.10.3

`{"level":"error","ts":1622139619.310684,"logger":"reconciler.vulnerabilityreport","msg":"Scan job container","job":"starboard-operator/scan-vulnerabilityreport-5b69f79cc","container":"kube-rbac-proxy-self","status.reason":"Error","status.message":"2021-05-27T18:20:17.861Z\t\u001b[31mFATAL\u001b[0m\tunable to initialize a scanner: unable to initialize a docker scanner: 3 errors occurred:\n\t* unable to inspect the image (quay.io/coreos/kube-rbac-proxy:v0.4.1): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n\t* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory\n\t* unsupported MediaType: \"application/vnd.docker.distribution.manifest.v1+prettyjws\", see https://github.com/google/go-containerregistry/issues/377\n\n\n","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/home/runner/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132\ngithub.com/aquasecurity/starboard/pkg/operator/controller.(*VulnerabilityReportReconciler).processFailedScanJob\n\t/home/runner/work/starboard/starboard/pkg/operator/controller/vulnerabilityreport.go:417\ngithub.com/aquasecurity/starboard/pkg/operator/controller.(*VulnerabilityReportReconciler).reconcileJobs.func1\n\t/home/runner/work/starboard/starboard/pkg/operator/controller/vulnerabilityreport.go:325\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/reconcile/reconcile.go:102\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/internal/controller/controller.go:263\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.7.2/pkg/internal/controller/controller.go:198\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.19.2/pkg/util/wait/wait.go:99"}`

Could you please help me to resolved this issue?

**Comment by danielpacak** [comment, marked as answer]
I think you have all the details in the error log.

```
unsupported MediaType: \"application/vnd.docker.distribution.manifest.v1+prettyjws\", see https://github.com/google/go-containerregistry/issues/377
```

Starboard integrates with Trivy static vulnerability scanner, which does not support (ancient) manifest `application/vnd.docker.distribution.manifest.v1+prettyjws`. Trivy relies on https://github.com/google/go-containerregistry/ module and the error log also points out to the GitHub issue, where you can find the whole discussion about v1+prettyjws manifest on https://github.com/google/go-containerregistry/issues/377

Just to confirm, the same behaviour is reproducible with Trivy run directly so there's nothing we can do about it in Starboard plugin for Trivy:

```console
$ trivy image quay.io/coreos/kube-rbac-proxy:v0.4.1
2021-05-28T09:22:15.419+0200	INFO	Need to update DB
2021-05-28T09:22:15.419+0200	INFO	Downloading DB...
21.52 MiB / 21.52 MiB [-----------------------------------------------------------------------------------------------------] 100.00% 47.22 MiB p/s 1s
2021-05-28T09:22:17.718+0200	FATAL	scan error: unable to initialize a scanner: unable to initialize a docker scanner: 3 errors occurred:
	* unable to inspect the image (quay.io/coreos/kube-rbac-proxy:v0.4.1): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
	* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory
	* unsupported MediaType: "application/vnd.docker.distribution.manifest.v1+prettyjws", see https://github.com/google/go-containerregistry/issues/377
```
